op-supervisor/supervisor/backend/syncnode/controller.go

```diff
@@ -29,17 +31,20 @@ type SyncNodesController struct {
 	backend backend
 
 	depSet depset.DependencySet
+
+	activationCheck *activation.Check
 }
 
 var _ event.AttachEmitter = (*SyncNodesController)(nil)
 
 // NewSyncNodesController creates a new SyncNodeController
-func NewSyncNodesController(l log.Logger, depset depset.DependencySet, eventSys event.System, backend backend) *SyncNodesController {
+func NewSyncNodesController(l log.Logger, depset depset.DependencySet, eventSys event.System, backend backend, activationCheck *activation.Check) *SyncNodesController {
 	return &SyncNodesController{
-		logger:   l,
-		depSet:   depset,
-		eventSys: eventSys,
-		backend:  backend,
+		logger:          l,
+		depSet:          depset,
+		eventSys:        eventSys,
+		backend:         backend,
+		activationCheck: activationCheck,
```

Comment:
Why do we pass the `activationCheck` here? Can't it just be recreated from the `depSet` and `logger`? Note also that I suggest we remove this type and just add the check function to the `DependencySet` struct.